src/DatePicker/DatePicker.tsx

```diff
@@ -52,7 +48,23 @@ export namespace DatePicker {
     className?: string;
     /** an optional style object to pass to top level element of the component */
     style?: React.CSSProperties;
-  };
+  } & (
+    | {
+        /** whether the user can remove the selected date */
+        isClearable: true;
+        /** current date */
+        value?: LocalDate;
```

Comment:
I'd prefer a `LocalDate | null` here, instead of `undefined`. This would be consistent with other components where the "empty" value is represented with `null`.